nameko/testing/pytest.py

```diff
@@ -287,7 +287,9 @@ def socket_creator():
             '127.0.0.1', web_config_port)
         gr = eventlet.spawn(ws_app.run_forever)
         active_sockets.append((gr, ws_app))
-        return wait_for_sock()
+        socket = wait_for_sock()
+        socket.app = ws_app
```

Comment:
`make_virtual_socket` instantiates the `socket` object and the `ws_app` object, and returns both. I was suggesting to make the `app` an explicit member of the `Socket` class, rather than attaching it here.

I also think `socket_creator` is also a confusing name, but let's not muddy the waters of this PR.